Content.Server/Light/EntitySystems/HandheldLightSystem.cs

```diff
@@ -0,0 +1,284 @@
+            // Thus we'll just check if the level changes.
+            if (component.Cell == null)
+                return null;
+
+            var currentCharge = component.Cell.CurrentCharge;
+
+            if (MathHelper.CloseToPercent(currentCharge, 0) || component.Wattage > currentCharge)
+                return 0;
+
+            return (byte?) ContentHelpers.RoundToNearestLevels(currentCharge / component.Cell.MaxCharge * 255, 255, SharedHandheldLightComponent.StatusLevels);
+        }
+
+        private void OnInit(EntityUid uid, HandheldLightComponent component, ComponentInit args)
+        {
+            EntityManager.EnsureComponent<PointLightComponent>(uid);
+            component.CellSlot = EntityManager.EnsureComponent<PowerCellSlotComponent>(uid);
+
+            // Want to make sure client has latest data on level so battery displays properly.
+            component.Dirty(EntityManager);
+        }
+
+        private void OnRemove(EntityUid uid, HandheldLightComponent component, ComponentRemove args)
+        {
+            HandleDeactivate(component);
+        }
+
+        private void OnInteractUsing(EntityUid uid, HandheldLightComponent component, InteractUsingEvent args)
+        {
+            if (args.Handled) return;
+
+            if (!_blocker.CanInteract(args.User)) return;
+            if (!component.CellSlot.InsertCell(args.Used)) return;
+            component.Dirty(EntityManager);
+            args.Handled = true;
+        }
+
+        private void OnUse(EntityUid uid, HandheldLightComponent component, UseInHandEvent args)
+        {
+            if (args.Handled) return;
+
+            if (ToggleStatus(args.User, component))
+                args.Handled = true;
+        }
+
+        /// <summary>
+        ///     Illuminates the light if it is not active, extinguishes it if it is active.
+        /// </summary>
+        /// <returns>True if the light's status was toggled, false otherwise.</returns>
+        public bool ToggleStatus(EntityUid user, HandheldLightComponent component)
+        {
+            if (!_blocker.CanUse(user)) return false;
+            return component.Activated ? TurnOff(component) : TurnOn(user, component);
+        }
+
+        private void OnExamine(EntityUid uid, HandheldLightComponent component, ExaminedEvent args)
+        {
+            args.PushMarkup(component.Activated
+                ? Loc.GetString("handheld-light-component-on-examine-is-on-message")
+                : Loc.GetString("handheld-light-component-on-examine-is-off-message"));
+        }
+
+        public override void Shutdown()
+        {
+            base.Shutdown();
+            _activeLights.Clear();
+        }
+
+        private void HandleActivate(HandheldLightComponent component)
+        {
+            _activeLights.Add(component);
+        }
+
+        private void HandleDeactivate(HandheldLightComponent component)
+        {
+            _activeLights.Remove(component);
+        }
```

Comment:
So these are just a relic from the component when it was using `(De)ActivateHandheldLightMessage`. Given that these functions are private and are only ever called in one place in this system, why not just remove them altogether now?